calicoctl/commands/node/diags.go

```diff
@@ -155,7 +160,43 @@ such as transfer.sh using curl or similar.  For example:
 
 }
 
-// writeDiags executes the dignostic commans and outputs the result in the file
+// getDockerLogs will attempt to grab logs for any "calico" named containers for hosted installs.
+func getDockerLogs(logDir string) {
+	os.Mkdir(logDir, os.ModeDir)
+
+	fmt.Println("Copying logs from Calico containers")
+	// Get a list of Calico containers running on this Node.
+	var stderr bytes.Buffer
+	cmd := exec.Command("docker", "ps", "-a", "--filter", "\"name=calico\"", "--format", "{{.Names}}")
+	cmd.Stderr = &stderr
+	out, err := cmd.Output()
+	if err != nil {
+		fmt.Printf("Could not run docker command: %s\n", stderr.String())
+		return
+	}
+
+	// No Calico containers found.
+	if string(out) == "" {
+		log.Debug("Did not find any Calico containers")
+		return
+	}
+
+	// Grab the log for each container and write it as <containerName>.log.
+	scanner := bufio.NewScanner(strings.NewReader(string(out)))
+	for scanner.Scan() {
+		name := scanner.Text()
+		cLog, err := exec.Command("docker", "logs", name).Output()
+		if err != nil {
+			fmt.Printf("Could not pull log for container %s: %s\n", name, err)
```

Comment:
shouldn't we `continue` here if we get an error from line 188? else it will try to access `cLog` which might be a `nil` in line 192